### ISSUE TITLE
Platform intesisbox not ready yet: None; Retrying in background in 30 seconds

### ISSUE THREAD

**Issue by zuhair68** [jnimmo/hass-intesisbox#20]
I set up the integration in the configuration.yaml as follows:

climate:
  - platform: intesisbox
    host: 192.168.3.120

I get this message after restarting HA (all files are copied to [custom_components/intesisbox]

Logger: homeassistant.components.climate
Source: helpers/entity_platform.py:268
Integration: Climate (documentation, issues)
First occurred: 3:53:28 AM (1 occurrences)
Last logged: 3:53:28 AM

Platform intesisbox not ready yet: None; Retrying in background in 30 seconds

**Comment by rjmidau** [comment]
I also see this in my installation. It looks like in my environment there's a 1-2 second gap between establishing a connection to the IntesisBox, and receiving the feature and state information from it (which is required to set up the Platform). In this time Home Assistant tries to set up the platform and hits the following line:

https://github.com/jnimmo/hass-intesisbox/blob/9bb29798c98233744f54b1ba6a5998b893567006/custom_components/intesisbox/climate.py#L126

**Comment by jnimmo** [comment]
Hi both, please try out this integration instead, putting this directory into your custom_components. It merges support for IntesisHome and IntesisBox. 
https://github.com/jnimmo/home-assistant/tree/intesishome-config-flow/homeassistant/components/intesishome

**Comment by ManuelJimenezBuendia** [comment]
Hi, I'm trying [https://github.com/jnimmo/home-assistant/tree/intesishome-config-flow/homeassistant/components/intesishome](https://github.com/jnimmo/home-assistant/tree/intesishome-config-flow/homeassistant/components/intesishome) as you suggest, but I don't manage to set up the integration properly. I've copied the folder 'intesishome' from the URL inside my 'config/custom_components' folder.

My configuration.yaml is as follows:
climate:
  \- platform: intesishome
    device: intesisbox
    host: 192.168.80.121

But I get the following error checking the yaml:
Invalid config for [climate.intesishome]: extra keys not allowed @ data['host']. Got '<192.168.80.121>'
required key not provided @ data['password']. Got None
required key not provided @ data['username']. Got None
value must be one of ['IntesisHome', 'airconwithme'] for dictionary value @ data['device']. Got 'intesisbox'. (See ?, line ?).

Any ideas?
Thanks.

**Comment by jnimmo** [comment]
It sounds like it must not be loading the custom_component - as in this version username and password are optional. Have you got any other custom_components successfully loading from the folder?
I think you should also be able to set this version up using Config Flow, but only if it is successfully replacing the built in platform with this one of course.

**Comment by ManuelJimenezBuendia** [comment]
I have the same feeling that the component is not replacing the buil in platform, but I have hacs, scheduler an spotcast loading from the folder and working properly. 
Many thanks, I'll keep looking for a solution.

**Comment by jnimmo** [comment]
Worth just double checking your folder includes the config_flow.py file too (in that the correct version has been downloaded)

**Comment by ManuelJimenezBuendia** [comment]
Yes it does:
![image](https://user-images.githubusercontent.com/24292006/173939425-128e175d-b8ea-4992-a3b5-a340bef2bced.png)

**Comment by ManuelJimenezBuendia** [comment]
Might is has something to do with the strings.json file? I see it is different from the translations/en.json

**Comment by ManuelJimenezBuendia** [comment]
Finally I made the integration intesisbox work. I lacked adding the integration inside de 'integrations' card inside Settings.
I performed the following steps:
1. Copy intesisbox folder inside config/custom_components.
2. Restart HASS
3. Add configuration keys in configuration.yaml:
```
  climate:
  - platform: intesisbox
    host: 192.168.80.121
```
4. Restart HASS.
5. Go to Settings > Devices & Services > Integrations
6. Add new integration: Intesisbox and submit the host IP

**Comment by Marcotic** [comment]
hello. I tried the steps above. but i get a configuration error.
Any idea of what is hapening?
Thanks

**Comment by Makestudio20** [comment]
I tried adding this integration to make a local connection to an IntesisBox Wifi box. When installing this and adding it via integrations and filling in the host IP address the integrations keeps spinning and says "initializing".
If I add 

```
  climate:
  - platform: intesisbox
    host: 192.168.x.x
```
to my configuration.yaml it says in the log: Setup of platform intesisbox is taking longer than 60 seconds. Startup will proceed without waiting any longer.

Looks like this is not loading correctly. How can this be fixed? Would love to be able to control my wifi box locally instead of via the accloud

**Comment by jbergler** [comment]
I've been making a bunch of improvements in the past few days, which I've now released as v1.0.0 of this integration.
If you'd be so kind as to try again and share debug logging with something like the following in your config should the issue persist

```
logger:
  default: info
  logs:
    custom_components.intesisbox: debug
```
